api/proto/config/appconfig.proto

```diff
@@ -49,9 +50,16 @@ message AppInstanceConfig {
     // App Instance initialization configuration data provided by user
     // This will be used as "user-data" in cloud-init
     // Empty string will indicate that cloud-init is not required
+    // to be deprecated, use cipherTextUserData instead
     string userData = 11;
 
     // Config flag if the app-instance should be made accessible
     // through a remote console session established by the device.
     bool remoteConsole = 12;
+
+    // encrypted userData
+    bytes cipherTextUserData = 13;
+
+    // cipher information to handle encrption/decryption
+    // CipherInfo cInfo = 14;
```

Comment:
Instead of this pattern it makes a bit more sense to but the cipherText inside the CipherInfo thus a 
 CipherInfo cipherTextUserData
in here.

Also, if you accidentally decrypt using the wrong key you will still get a byte string. You need something like a sha to verify that the result is correct. Adding a clearTextSha256 field to the CipherInfo would be one way to handle that.